paper/src/main/java/sh/okx/railswitch/switches/SwitchLogic.java

```diff
@@ -0,0 +1,43 @@
+package sh.okx.railswitch.switches;
+
+import com.google.re2j.Matcher;
+import org.bukkit.entity.Player;
+
+/**
+ * Base class for switch decisions, and a function to determine which subclass to use
+ */
+public abstract class SwitchLogic {
+	public SwitchLogic() throws Exception {
+
+	}
+
+	abstract boolean decide(Player player) throws Exception;
+
+	public static SwitchLogic try_create(String[] lines) throws Exception {
+		//Simple destinations
+		if (SimpleDestLogic.isNormal(lines[0]) || SimpleDestLogic.isInverted(lines[0]))
+			return new SimpleDestLogic(lines);
+
+		//Regex destinations
+		Matcher destex = DestExLogic.DESTEX.matcher(lines[0]);
+		if (destex.matches()) return new DestExLogic(lines, destex);
+
+		//Destination adding
+		if (lines[0].equalsIgnoreCase(AddDestLogic.INVOCATION)) return new AddDestLogic(lines);
+
+		//Destination removal
+		if (lines[0].equalsIgnoreCase(RemoveDestLogic.INVOCATION)) return new RemoveDestLogic(lines);
+
+		//Destination removal via regex
+		Matcher destexrm = RemoveDestExLogic.DESTEX.matcher(lines[0]);
+		if (destexrm.matches()) return new RemoveDestExLogic(lines, destexrm);
+
+		//Failure to specify the [] bits
+		if (lines[0].length() >= 2 && lines[0].charAt(0) == '[' && lines[0].charAt(lines[0].length() - 1) == ']') {
```

Comment:
Can be replaced with [`lines[0].startsWith("[")`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/String.html#startsWith(java.lang.String)) and [`lines[0].endsWidth("]")`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/String.html#endsWith(java.lang.String))... also never `==` strings